### ISSUE TITLE
about the speed of tiny

### ISSUE THREAD

**Issue by wzjiang** [eric612/MobileNet-YOLO#33]
Hello, I train the yolov3 tiny(416*416) with my car plate detection dataset with your help successfully. But when I test it in 1080Ti with gpu，I found it is slow. It is about 0.15s/f . So I want to test the speed of every layer by run "./build/install/bin/caffe time --model=models/plate_detection/tiny-yolov3-deploy.prototxt --weights=models/plate_detection/models/deploy-_iter_30000.caffemodel --gpu=1". And there is a error such as 
-------------------------------------------------------------------------------------
F1116 13:21:08.834992 18187 yolov3_detection_output_layer.hpp:67] Not Implemented Yet
*** Check failure stack trace: ***
    @     0x7f4396d3cdaa  (unknown)
    @     0x7f4396d3cce4  (unknown)
    @     0x7f4396d3c6e6  (unknown)
    @     0x7f4396d3f687  (unknown)
    @     0x7f4397455dfb  caffe::Yolov3DetectionOutputLayer<>::Backward_cpu()
    @           0x40f16d  time()
    @           0x40b493  main
    @     0x7f4395aadf45  (unknown)
    @           0x40bdf6  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)
----------------------------------------------------------------------------------------
I checked the yolov3_detection_output_layer.hpp  and found there is cpu achievement codes without gpu mode codes . What could I do to accelerate this project( I remember the speed of yolov3 tiny with the darknet is about 3ms/f).

**Comment by wzjiang** [comment]
I run the project with cpu mode such as "./build/install/bin/caffe time --model=models/plate_detection/tiny-yolov3-deploy.prototxt --weights=models/plate_detection/models/deploy-_iter_30000.caffemodel". I found it have a same error.

**Comment by eric612** [comment]
* yolov3_detection_output_layer gpu code is likely [yolov3_layer.cu](https://github.com/eric612/MobileNet-YOLO/blob/master/src/caffe/layers/yolov3_layer.cu) , you can reference it to implement or wait me upload 
* merge batch norm and scale layer into conv. layer before inference , this step may increase speed
* the performance between darknet and this caffe version were not equal , you can find the suitable inference framework for you , like tensorRT , ncnn ...

**Comment by wzjiang** [comment]
Amazing！Did I make any mistakes?
I test the speed of your yolo v3 lite by running “sh demo_yolo_lite.sh” and the speed is about 0.011 s/f.
But test the speed of my yolov3 is about 0.15s/f( the prototxt is modified from https://drive.google.com/drive/folders/1RqiYrPvophkv72KNKdPkkcGEz5yqattv)

**Comment by greenteaofwhu** [comment]
hi,I used the yolov3_tiny.prototxt(include solver.prototxt,tiny-yolov3-train.prototxt,tiny-yolov3-test.prototxt)that you gave to train and test on the VOC, but after 80,000 iterations, the final map is only 0.313 and  I don't know if it is necessary to train on a pre-trained model. Can you give me Some suggestions to deal with this problem?

**Comment by eric612** [comment]
@greenteaofwhu ,
What pre-trained weights did you use ?  I suggest use the coco or darknet reference weights
